### PR TITLE
Tune process killing to make sure that xcodebuild does not leave unnecessary leftovers

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Sometimes there are strange side effects while executing xcodebuild. I assume this might happen because we always try to force kill the process, so there might be some unneeded leftovers in the system, like opened sockets. 
I've also added the verification of the fact, that the xcodebuild process is properly killed before simulator shutdown.

### CODE REVIEW

**Comment by saikrishna321** [comment]
i guess jliips is on vacation. @imurchie can this be reviewed

File: lib/driver.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ import { detectUdid, getAndCheckXcodeVersion, getAndCheckIosSdkVersion,
          adjustWDAAttachmentsPermissions, checkAppPresent, getDriverInfo,
          clearSystemFiles, translateDeviceName, normalizeCommandTimeouts,
          DEFAULT_TIMEOUT_KEY } from './utils';
+import { killAppUsingAppName } from './wda/utils';
```
Comment (imurchie) [inline lib/driver.js]: If this is being used outside of the `wda` folder it should be moved into the root `utils.js`.

**Comment by mykola-mokhnach** [comment]
Moved the stuff to the main utils module

**Comment by imurchie** [comment]
Awesome! Thanks!

**Comment by imurchie** [comment]
Published in `appium-xcuitest-driver@2.43.0`.

**Pull request merged**